code/modules/mapping/space_management/multiz_helpers.dm

```diff
@@ -7,6 +7,19 @@
 		return get_step(SSmapping.get_turf_below(get_turf(ref)), dir)
 	return get_step(ref, dir)
 
+/proc/get_multiz_accessible_levels(center_z)
+	. = list(center_z)
+	var/other_z = center_z
+	var/offset
+	while((offset = SSmapping.level_trait(other_z, ZTRAIT_DOWN)))
+		other_z += offset
+		. += other_z
+	other_z = center_z
+	while((offset = SSmapping.level_trait(other_z, ZTRAIT_UP)))
+		other_z += offset
+		. += other_z
+	return .
+
```

Comment:
Suffering, but necessary suffering. This does mean that looping z levels will crash the  game, but I don't care.